src/pal/src/map/map.cpp

```diff
@@ -2445,7 +2439,7 @@ void * MAPMapPEFile(HANDLE hFile)
     // First try to reserve virtual memory using ExecutableAllcator. This allows all PE images to be
     // near each other and close to the coreclr library which also allows the runtime to generate
     // more efficient code (by avoiding usage of jump stubs).
-    loadedBase = ReserveMemoryFromExecutableAllocator(pThread, virtualSize);
+    loadedBase = ReserveMemoryFromExecutableAllocator(pThread, ((virtualSize-1) & ~VIRTUAL_PAGE_MASK) + VIRTUAL_PAGE_SIZE);
```

Comment:
This calculation is incorrect. The allocated size needs to be the size of the virtual address space range that is needed to map the whole executable file on the current system. That can be more than just the virtualSize aligned up to the page size (each section mapped with adjustment requires one more page of virtual memory)